src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/LegacyPackageReferenceProject.cs

```diff
@@ -179,50 +180,64 @@ public override async Task<IEnumerable<PackageReference>> GetInstalledPackagesAs
             return packages.InstalledPackages;
         }
 
-        /// <summary>
-        /// Gets the both the installed (top level) and transitive package references for this project.
-        /// Returns the package reference as two separate lists (installed and transitive).
-        /// </summary>
+        /// <inheritdoc/>
         public override async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsync(CancellationToken token)
         {
-            PackageSpec packageSpec = await GetPackageSpecAsync(NullSettings.Instance);
-
-            var frameworkSorter = new NuGetFrameworkSorter();
-
-            string assetsFilePath = await GetAssetsFilePathAsync();
-            var fileInfo = new FileInfo(assetsFilePath);
-            IList<LockFileTarget> targets = default;
-
-            if (fileInfo.Exists && fileInfo.LastWriteTimeUtc > _lastTimeAssetsModified)
+            PackageSpec reading = await GetCachedPackageSpecAsync(token);
+            if (reading! == null)
```

Comment:
This logic contradicts CPSPackageReferenceProject.cs one. Why?
`if (reading == null)`

Reply:
Fixed.